topi/python/topi/nn/softmax.py

```diff
@@ -4,28 +4,49 @@
 import tvm
 
 @tvm.tag_scope(tag='softmax_output')
-def softmax(x):
+def softmax(x, axis=-1):
     """Perform softmax activation on the data
 
     Parameters
     ----------
     data : tvm.Tensor
-        2-D input data
+        can be any dimension
+
+    axis : int
+        channel axis
 
     Returns
     -------
     output : tvm.Tensor
-        2-D output with same shape
+        output shape is the same as input
     """
-    assert len(x.shape) == 2, "only support 2-dim softmax"
-    m, n = x.shape
-    k = tvm.reduce_axis((0, n), name='k')
-    max_elem = tvm.compute((m, ), lambda i: tvm.max(x[i, k], axis=k))
-    k = tvm.reduce_axis((0, n), name='k')
-    expsum = tvm.compute(
-        (m, ), lambda i: tvm.sum(tvm.exp(x[i, k] - max_elem[i]), axis=k))
-    return tvm.compute(
-        x.shape, lambda i, j: tvm.exp(x[i, j] - max_elem[i]) / expsum[i])
+    assert axis < len(x.shape)
+    shape = x.shape
+    if axis == -1:
```

Comment:
seems -1 is specifically handled, since we support negative indexing convention. I think we should do
```python
if axis < 0:
    axis = len(x.shape) + axis
```